nion/instrumentation/camera_base.py

```diff
@@ -565,7 +565,9 @@ def __init__(self):
         self.name = None
         self.uuid = uuid.uuid4()
 
-    def add_layer(self, graphic: Graphics.Graphic, value: typing.Union[float, str], inverted: bool = False):
+    def add_layer(self, graphic: Graphics.Graphic, value: typing.Union[float, numpy.ndarray], inverted: bool = False):
```

Comment:
Shouldn't the type on `value` be `typing.Union[float, str, numpy.ndarray]`?

Reply:
Also, can you add a test for this (and other mask variations)? Tests will be helpful for the upcoming acquisition pipeline code.

See `test_grab_synchronized_camera_data_channel_basic_sum_masked`.

Reply:
> 
> 
> Shouldn't the type on `value` be `typing.Union[float, str, numpy.ndarray]`?

My idea was to deprecate the string values ("grad_x" and "grad_y") because they are highly specific. I just wanted to keep support for those values to not break existing masks. But when creating a new maks (which is the situation when you would call `add_layer`) you should not use "grad_x" and "grad_y" anymore.

Reply:
> 
> 
> Also, can you add a test for this (and other mask variations)? Tests will be helpful for the upcoming acquisition pipeline code.
> 
> See `test_grab_synchronized_camera_data_channel_basic_sum_masked`.

Do you mean a test for the mask class itself or for masked summing in a synchronzed acqusition?

Reply:
For the typing: my plan is to have everything pass mypy checks. To do that, it will either require a different type at the argument level; or type casting at the place where strings are passed and where they're checked within the code (around line 604-610). So maybe just add the explicit `typing.cast(str, value)` on lines 606 and 608?

For the tests: for masked summing (similar to `test_grab_synchronized_camera_data_channel_basic_sum_masked`). I'd like to have a way to ensure this continues working within the new acquisition pipeline.